parse/parse.ts

```diff
@@ -0,0 +1,72 @@
+import type { Reader } from "../deps.ts";
+import { readLines } from "../deps.ts";
+import { quotedSplit } from "./quoted_split.ts";
+
+/**
+ * ParsedComment is a command that is generated by a //deno:generate
+ * comment.
+ */
+export interface ParsedComment {
+  args: string[];
+  original: string;
+  line: bigint;
+  character: bigint;
+  alias?: string;
+}
+
+/**
+ * DIRECTIVE_PATTERN is the pattern that matches a //deno:generate comment.
+ */
+export const DIRECTIVE_PATTERN = /^\/\/deno:generate\s+/;
+
+/**
+ * This is a very simple parser that only looks for comments
+ * because to run arbitrary code generation commands, we only
+ * need to parse the file to determine the comments. With each
+ * comment, we return the line and character of the comment so that
+ * we can tell the user where errors are.
+ *
+ * Implements dynamic programming to get the line and character of the comment.
+ */
+export async function parseComments(reader: Reader): Promise<ParsedComment[]> {
+  const comments: ParsedComment[] = [];
+  let currentLine = BigInt(0);
```

Comment:
BigInt is probably unnecessary here.  You're unlikely to ever parse a file with more than 9007199254740991 lines. And if you care about that case, you might just want to skip files that are over that many bytes long. 😊

Reply:
> BigInt is probably unnecessary here. You're unlikely to ever parse a file with more than 9007199254740991 lines. And if you care about that case, you might just want to skip files that are over that many bytes long. 😊

I agree that using BigInt is unnecessary in this context. Considering the extremely rare case of parsing a file with more than 9007199254740991 lines, it's reasonable to skip files that exceed that length. I have gone ahead and committed the suggested changes <https://github.com/EthanThatOneKid/deno_generate/pull/1/commits/dffc7a0213afc3db91007f0b9693df72f93a0ec9>.